packages/@react-aria/collections/src/Document.ts

```diff
@@ -344,6 +344,9 @@ export class ElementNode<T> extends BaseNode<T> {
     node.rendered = rendered;
     node.render = render;
     node.value = value;
+    if (obj['aria-label']) {
```

Comment:
generally we try to add as little as possible to these nodes, so I think we'd prefer the other option and read from `focusedItem.props['aria-label']` and `section.props['aria-label']`

Reply:
Alright. Just to clarify, the attribute `aria-label` already on the node, it is just not initialized:

https://github.com/adobe/react-spectrum/blob/307b71c61074c33fae6e9eb79d3d29b9c83150a0/packages/%40react-aria/collections/src/BaseCollection.ts#L23-L42

Would you still prefer the other way (read from `focusedItem.props['aria-label']` and `section.props['aria-label']`)?

Reply:
Just a small note, I removed the if condition i originally had here to ensure that the sync performed by `setProps` also works if the react-aria prop changes from a value to `undefined`.

See https://github.com/adobe/react-spectrum/compare/5a6251896b55a27807bef2630208bfb96a6657df..a11dcf8bb46322cad7d775e3ff27e98b0fc6667f#diff-11dea92f1a0629c980fc22a1e00bf0cebed068aff25abf42c47264618fce3cc6

Reply:
thanks for the update and the additional context 👍  appreciate it.